apps/cyberstorm-remix/cyberstorm/utils/StrongForm/useStrongForm.ts

```diff
@@ -45,112 +126,156 @@ export function useStrongForm<
   const [submitError, setSubmitError] = useState<SubmissionError>();
   const [inputErrors, setInputErrors] = useState<InputErrors>();
 
-  useEffect(() => {
-    if (refining || submitting) {
-      return;
+  const ensureSubmissionDataShape = (value: Inputs): SubmissionDataShape => {
+    if (
+      value === null ||
+      (typeof value !== "object" && typeof value !== "function")
+    ) {
+      throw new Error(
+        "useStrongForm received primitive form inputs without a refiner; provide a refiner or ensure the input type matches the submission data shape."
+      );
+    }
+
+    return value as SubmissionDataShape;
+  };
+
+  const defaultErrorMapper = (error: unknown): UserFacingError => {
+    if (error instanceof UserFacingError) {
+      return error;
     }
+    return mapApiErrorToUserFacingError(error);
+  };
+
+  const mapError: (error: unknown) => SubmissionError =
```

Comment:
My IDE flags this for type error, and I don't know why this variable is even used since it's called in only one place, and why it's defined here since the call site comes way later.